lib/puma/const.rb

```diff
@@ -147,15 +149,7 @@ module Const
     MAX_BODY = MAX_HEADER
 
     REQUEST_METHOD = "REQUEST_METHOD".freeze
-    HEAD = "HEAD".freeze
-    GET = "GET".freeze
-    POST = "POST".freeze
-    PUT = "PUT".freeze
-    DELETE = "DELETE".freeze
-    OPTIONS = "OPTIONS".freeze
-    TRACE = "TRACE".freeze
-    PATCH = "PATCH".freeze
-    SUPPORTED_HTTP_METHODS = [HEAD, GET, POST, PUT, DELETE, OPTIONS, TRACE, PATCH].freeze
+    DEFAULT_SUPPORTED_HTTP_METHODS = %w[HEAD GET POST PUT DELETE OPTIONS TRACE PATCH].to_set.freeze
```

Comment:
This files specifies

    # frozen_string_literal: true

so why freeze strings ?

Reply:
See https://github.com/puma/puma/commit/88e51fb08e0735a98a519db46649f01bcc88d03c, the magic comment was added when Puma still supported Ruby versions not supporting the magic comment

Reply:
That's what I thought, thanks for pointing to the historical context. :+1: